src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2798,11 +2797,18 @@ else if (Gossiper.instance.compareEndpointStartup(endpoint, existing) > 0)
 
         // capture because updateNormalTokens clears moving and member status
         boolean isMember = getTokenMetadata().isMember(endpoint);
-        boolean isMoving = getTokenMetadata().isMoving(endpoint);
+        boolean isMoving = getTokenMetadata().isMoving(endpoint) || operationMode == Mode.MOVING;
+
+        if (!isMember && !isMoving)
```

Comment:
shouldn't it be isMember rather than !isMember?

Reply:
it is correct, we want to run this code only if we are joining that node to the cluster. `isMember = false` for a joining node.